src/astral/data-utils.js

```diff
@@ -0,0 +1,86 @@
+const tokenCache = {
+    access: null,
+    apiKey: null,
+    refresh: null,
+    expires: null
+};
+
+const _getTokenFromDb = () => new Promise((resolve, reject) => {
+    const req = indexedDB.open("firebaseLocalStorageDb", 1);
+    console.log("Created db req", req)
+    req.onsuccess = (event) => {
+        const db = req.result;
+        const trans = db.transaction("firebaseLocalStorage", IDBTransaction.READ)
+        const objStore = trans.objectStore("firebaseLocalStorage");
+
+        const cursorReq = objStore.openCursor();
+
+        cursorReq.onsuccess = (event) => {
+            console.log("Create cursor req", cursorReq.result)
+            resolve(cursorReq.result.value.value.stsTokenManager);
+        };
```

Comment:
Is there a onerror for cursorReq too ? and can you check if the `result.value.value.stsTokenManager` does exist ?